synapseclient/models/permission.py

```diff
@@ -0,0 +1,175 @@
+from dataclasses import dataclass
```

Comment:
For this `permission.py` let's create a new MD page and nav link so it'll show up in the documents pages.

Note on running the doc pages locally: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/CONTRIBUTING.md#python-doc-pages

Where you'll be adding the MD page to: https://github.com/Sage-Bionetworks/synapsePythonClient/tree/develop/docs/reference

Where you'll add it so that it shows up in the nav panel: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/mkdocs.yml#L53 - Let's add this above `Core` for now

Reply:
<img width="1151" alt="Screenshot 2023-12-28 at 10 57 53 AM" src="https://github.com/Sage-Bionetworks/synapsePythonClient/assets/90745557/dda0435b-a2fe-4415-9dbb-dfcf6273779f">